app/controllers/products_controller.rb

```diff
@@ -0,0 +1,59 @@
+class ProductsController < ApplicationController
+  def index
+    @products = Product.all
+  end
+
+  def show
+    @product = Product.find(params[:id])
```

Comment:
@product = Product.find(params[:id])

this line can be in a before_action since it is being used in multiple actions